src/App.tsx

```
@@ -1,26 +1,47 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { useEffect, useState } from 'react';
+import { getTodos } from './api';
+import { todosSlice } from './features/todos';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<string | null>(null);
+
+  useEffect(() => {
+    setError(null);
+    setLoading(true);
+
+    getTodos()
+      .then(todos => dispatch(todosSlice.actions.setTodos(todos)))
+      .catch(() => {
```

Error handling: the catch block swallows the actual error. Change the catch to accept an error parameter and log it (e.g. `catch(err => { console.error(err); setError('Something went wrong'); })`) so failures are visible during debugging. The existing code sets a user-facing message which is fine, but console logging the error will help during development and CI troubleshooting.